tools/snp.sh

```diff
@@ -164,6 +172,83 @@ verify_snp_host() {
   fi
 }
 
+verify_platform_snp_bit_status() {
+  if [ "$1" == "host" ]; then
+    # Get the host cpuid eax
+    local host_cpuid_eax
+    host_cpuid_eax=$(cpuid -1 -r -l 0x8000001f | grep -oE "eax=[[:alnum:]]+ " | cut -c5- | tr -d '[:space:]')
+
+    # Map all the computed host sev/snp bit values in a single associative array
+    declare -A actual_sev_snp_bit_status=(
+      [SME]=$((${host_cpuid_eax} & 1))
+      [SEV]=$((${host_cpuid_eax} & 2))
+      [SEV-ES]=$((${host_cpuid_eax} & 8))
+      [SNP]=$((${host_cpuid_eax} & 16))
+    )
+
+    # Map all the expected host sev/snp bit values in a single associative array
+    declare -A expected_sev_snp_bit_value=(
+      [SME]=1
+      [SEV]=2
+      [SEV-ES]=8
+      [SNP]=16
+      )
+  fi
+
+  if [ "$1" == "guest" ]; then
+    if [ ! -f "${GUEST_SSH_KEY_PATH}" ]; then
+      >&2 echo -e "Guest SSH key not present [${GUEST_SSH_KEY_PATH}], so cannot verify guest SNP enabled"
+      return 1
+    fi
+
+    # Install guest rdmsr package dependencies to insert & insert guest msr module
+    ssh_guest_command "sudo DEBIAN_FRONTEND=noninteractive sudo apt install -y msr-tools > /dev/null 2>&1" > /dev/null 2>&1
+    ssh_guest_command "sudo modprobe msr" > /dev/null 2>&1
+
+    # Read the guest (MSR_AMD64_SEV) value
+    local guest_msr_read
+    guest_msr_read=$(ssh_guest_command "sudo rdmsr -p 0 0xc0010131")
+    guest_msr_read=$(echo "${guest_msr_read}" | tr -d '\r' | bc)
+
+    # Map all the sev features in a single associative array for all guest SEV features
+    declare -A actual_sev_snp_bit_status=(
+      [SEV]=$((${guest_msr_read} & 1))
+      [SEV-ES]=$((${guest_msr_read} & 2))
+      [SNP]=$((${guest_msr_read} & 4))
+    )
+
+    declare -A expected_sev_snp_bit_value=(
+      [SEV]=1
+      [SEV-ES]=2
+      [SNP]=4
+    )
```

Comment:
Same thing as above. You can just get the specific bit for each feature and that way you don't need 2 different arrays.